### PR TITLE
Gh 1434/display raw transaction data in advanced transaction details

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #1434

### CODE REVIEW

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 1]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+    }
+
+    func buildSections(_ tx: SCGModels.TransactionDetails) {
+        if let hash = tx.txHash?.description {
+            sections.append(Section(title: "Chain transaction data",
+                                    items: [SectionItem(title: "Transaction hash:", value: hash)]))
+        }
+
+        if let txData = tx.txData {
+            var safeTransactionData:[SectionItem] = []
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]:
```suggestion
            var safeTransactionData: [SectionItem] = []
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 2]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+            safeTransactionData.append(SectionItem(title: "To:", value: txData.to))
+            safeTransactionData.append(SectionItem(title: "Value:", value: txData.value.description))
+            if let data = txData.hexData {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift:65]:
```suggestion

            if let data = txData.hexData {
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 3]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+        } else {
+            nonce = nil
+            safeTxHash = nil
+        }
+
+        if let nonce = nonce, let safeTxHash = safeTxHash {
+            sections.append(Section(title: "",
+                                    items: [SectionItem(title: "safeTxHash:", value: safeTxHash),
+                                            SectionItem(title: "Nonce:", value: nonce)]))
+        }
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]: these constants are effectively used only for the case when they exist, so I'd refactor it this way:

```suggestion
        if case SCGModels.TransactionDetails.DetailedExecutionInfo.multisig(let multisigTx)? = tx.detailedExecutionInfo {
            sections.append(Section(title: "",
                                    items: [SectionItem(title: "safeTxHash:", value: multisigTx.safeTxHash.description),
                                            SectionItem(title: "Nonce:", value: multisigTx.nonce.description)]))
        }
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 4]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+            }
+
+            sections.append(Section(title: "Signatures", items: signatures))
+        case .module(let moduleInfo):
+            sections.append(Section(title: "Module Data", items: [SectionItem(title: "Module Address:", value: moduleInfo.address)]))
+        default:
+            break
+        }
+
+        sections = sections.filter({ section in !section.items.isEmpty })
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]:
```suggestion
        sections = sections.filter { section in !section.items.isEmpty }
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 5]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        sections.count
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        items.count
+        sections[section].items.count
+    }
+
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection _section: Int) -> UIView? {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]:
```suggestion
    override func tableView(_ tableView: UITableView, viewForHeaderInSection sectionIndex: Int) -> UIView? {
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 6]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+        let section = sections[_section]
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]:
```suggestion
        let section = sections[sectionIndex]
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 7]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
+        let view = tableView.dequeueHeaderFooterView(BasicHeaderView.self)
+        view.setName(section.title)
+
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]:
```suggestion
```

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift [hunk 8]
```diff
@@ -28,30 +29,146 @@ class AdvancedTransactionDetailsViewController: UITableViewController {
...
-            cell.setAccount(address: addressInfo.value.address, label: addressInfo.name, title: item.title, imageUri: addressInfo.logoUri)
-            return cell
-        } else if let string = item.value as? String {
-            let cell = tableView.dequeueCell(DetailExpandableTextCell.self, for: indexPath)
-            cell.setTitle(item.title)
-            cell.setCopyText(string)
-            cell.setText(string)
-            cell.setExpandableTitle(nil)
-            return cell
+            return address(addressInfo.value.address, label: addressInfo.name, title: item.title, imageUri: addressInfo.logoUri, indexPath: indexPath)
```
Comment (DmitryBespalov) [inline Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift]:
```suggestion
            return address(addressInfo.value.address, 
                                      label: addressInfo.name, 
                                      title: item.title, 
                                      imageUri: addressInfo.logoUri, 
                                      indexPath: indexPath)
```

**Review by DmitryBespalov — COMMENTED** [review]
refactoring for better readability is needed

File: Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift (hunk 3 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Transaction/TransactionDetailsViewController/AdvancedTransactionDetailsViewController.swift:80]: This way the section will be always there even if there is no value for both, I have refactored it in different way

**Review by sche — APPROVED** [review]

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**